tests/Simplify/EqualTest.elm

```diff
@@ -934,6 +934,587 @@ a = ({ a = 1 }).a == ({ a = 1 }).b
+a : Bool
+a =  (l |> (not List.isEmpty))
+"""
+                        ]
+        ]
+
+
+setTests : Test
+setTests =
+    describe "Set.size should be Set.isEmpty"
+        [ test "should replace Set.size s == 0 with Set.isEmpty s" <|
+            \() ->
+                """module A exposing (..)
+a : Bool
+a = Set.size s == 0
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "This can be replaced with a call to Set.isEmpty"
+                            , details =
+                                [ "Set.size takes as long to run as the list is long"
+                                , "whereas Set.isEmpty runs in constant time."
+                                ]
+                            , under = "Set.size s == 0"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a : Bool
+a = Set.isEmpty s 
+"""
+                        ]
+        , Test.skip <|
+            test "should replace size s == 0 with isEmpty s" <|
+                \() ->
+                    """module A exposing (..)
+import Set as CoreSet exposing (..)
+
+
+a : Bool
+a = size s == 0
+"""
+                        |> Review.Test.run ruleWithDefaults
+                        |> Review.Test.expectErrors
+                            [ Review.Test.error
+                                { message = "This can be replaced with a call to Set.isEmpty"
+                                , details =
+                                    [ "Set.size takes as long to run as the list is long"
+                                    , "whereas Set.isEmpty runs in constant time."
+                                    ]
+                                , under = "size s == 0"
+                                }
+                                |> Review.Test.whenFixed """module A exposing (..)
+import Set as CoreSet exposing (..)
+
+
+a : Bool
+a = isEmpty s 
+"""
+                            ]
+        , test "should replace 0 == Set.size s with Set.isEmpty s" <|
+            \() ->
+                """module A exposing (..)
+a : Bool
+a = 0 == Set.size s
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "This can be replaced with a call to Set.isEmpty"
+                            , details =
+                                [ "Set.size takes as long to run as the list is long"
+                                , "whereas Set.isEmpty runs in constant time."
+                                ]
+                            , under = "0 == Set.size s"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a : Bool
+a =  Set.isEmpty s
```

Comment:
I was torn about spacing here. When I do my text replacement I could try to move one to the left in the range, but that assumes that the code is `elm-format`ted. Is there a canonical way to replace this? Am

Reply:
This is honestly fine. If you really want to improve it, you probably should:
- if `List.length l == 0`: remove from the end of `List.length` until the end of `0`
- if `0 == List.length l`: remove from the start of `0` until the start of `List.length`

and then also replace `List.length` by the appropriate function. Let me know if I misunderstood your question.